File: src/problem5.js
"use strict";

function problem() {
    let result = 1;
    for (let i = 2; i <= 20; i++) {
        result = findPPCM(result, i);
    }
    return result;


    function findPPCM(a, b) {
        return a * b / findPGCD(a, b);
    }

    function findPGCD(a, b) {
        if (a > b) {
            const c = b;
            b = a;
            a = c;
        }

        let remainder = b % a;
        while (remainder != 0) {
            b = a;
            a = remainder;
            remainder = b % a;
        }
        return a;
    }
}

module.exports = problem;

File: java/src/main/java/com/oltruong/Problem5.java
package com.oltruong;

class Problem5 {

    private static final int NUMBER = 20;

    static long getResult() {
        long result = 1;
        for (long i = 2; i <= NUMBER; i++) {
            result = findPPCM(result, i);
        }

        return result;
    }

    private static long findPPCM(long a, long b) {
        return a * b / findPGCD(a, b);
    }

    private static long findPGCD(long a, long b) {
        if (a > b) {
            return findPGCD(b, a);
        } else {
            long remainder = b % a;
            if (remainder == 0) {
                return a;
            } else {
                return findPGCD(remainder, a);
            }
        }

    }
}
